packages/app/src/cli/utilities/app/http-reverse-proxy.ts

```diff
@@ -67,14 +53,38 @@ export async function runConcurrentHTTPProcessesAndPathForwardTraffic(
 
   const availablePort = portNumber ?? (await port.getRandomPort())
 
+  const proxy = httpProxy.createProxy()
+  const server = http.createServer(function (req, res) {
+    const target = match(rules, req)
+    if (target) return proxy.web(req, res, {target})
+
+    res.statusCode = 500
+    res.end('The request url and path did not match any of the listed rules!')
+  })
+
+  // Capture websocket requests and forward them to the proxy
+  server.on('upgrade', function (req, socket, head) {
```

Comment:
Is there any corresponding event we need to look at to stop the proxy? Or this is enough?

Reply:
nop, the `upgrade` is an http request to upgrade from http to ws. From there everything is handled by the websocket server.